Enh/update xsd

bzah opened the pull request:
- [ ] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.


This PR fixes #61 

- Updated XSD schema to latest netCDF-java version from https://github.com/Unidata/netcdf-java/blob/maint-5.x/cdm/core/src/main/resources/resources/nj22/schemas/ncml-2.2.xsd#L4
- Updated dataclasses using https://pypi.org/project/xsdata/  (I had to manually modify the result to match the current syntax though)
- Fix issue with scalar parsing which value was not populated in the resulting DataArray.

huard reviewed (approved):
I suspect the scalar handling will break code that previously expected sequences for everything. Please mention it loud and clear in the CHANGELOG.

bzah commented:
Yes and it might break when the `values.content` is empty too.
Maybe I should split that into another Issue and PR.

bzah commented:
I refactored the scalar handling to included more cases and added related integration tests. I also documented the changes but prior to 0.4.0, scalar variables were not handled at all so it's not really a breaking changes more a new feature.
